apis/management.cattle.io/v3/pipeline_types.go

```diff
@@ -89,14 +89,16 @@ type PipelineStatus struct {
 }
 
 type PipelineSpec struct {
-	ProjectName string `json:"projectName" yaml:"projectName" norman:"required,type=reference[project]"`
+	ProjectName string `json:"projectName,omitempty" yaml:"-" norman:"required,type=reference[project]"`
 
-	DisplayName           string `json:"displayName,omitempty" yaml:"displayName,omitempty" norman:"required"`
+	DisplayName           string `json:"displayName,omitempty" yaml:"name,omitempty"`
```

Comment:
In the framework we do translations between API and store layer.
For example on creation we get `name` field, walk through mappers of the schema, and change it to `displayName` to store.
On queries we get `displayName` in CRD, then turn it to `name` in the APIs.
When importing the yaml file I want to obey the framework so that the validator/generator is the same as the creation from creation API. But I don't want to walk through mappers in the schema when exporting the yaml file because some mappers are adding unneccesary fields like `type`.
So I do convertion to the API fields like `name` directly by yaml annotations